hare/algorithm.go

```diff
@@ -47,16 +54,57 @@ type State struct {
 	certificate *pb.Certificate // the certificate
 }
 
+type StateQuerier interface {
+	IsIdentityActive(id []byte) bool
+}
+
+type Msg struct {
+	*pb.HareMessage
+	PubKey []byte
+}
+
+type mockStateQuerier struct {
+}
+
+func (msq mockStateQuerier) IsIdentityActive(id []byte) bool {
+	return true
+}
+
+func newMsg(hareMsg *pb.HareMessage, querier StateQuerier) (*Msg, error) {
+	// data msg to bytes
+	data, err := proto.Marshal(hareMsg.Message)
+	if err != nil {
+		log.Error("Could not marshal err=%v", err)
+		return nil, err
+	}
+
+	// extract pub key
+	pubKey, err := ed25519.ExtractPublicKey(data, hareMsg.InnerSig)
+	if err != nil {
+		log.Error("newMsg constrcution failed: err=%v", err, len(hareMsg.InnerSig))
+		return nil, err
+	}
+
+	// check identity is active
+	pub := signing.NewPublicKey(pubKey)
+	if !querier.IsIdentityActive(pub.Bytes()) {
```

Comment:
`IsIdentityActive` should return an error. We should be able to distinguish between the case where the user doesn't exist (i.e. broken signature) and the case where the user exists but isn't active. Once you get an error just print it to the log and return it

Reply:
Fixed